engine/api/src/org/apache/cloudstack/engine/cloud/entity/api/VirtualMachineEntity.java

```diff
@@ -115,6 +115,12 @@ String reserve(DeploymentPlanner plannerToUse, @BeanParam DeploymentPlan plan, E
     boolean stop(String caller) throws ResourceUnavailableException, CloudException;
 
     /**
+     * Stop the virtual machine, optionally force
+     *
+     */
+    boolean stop(String caller, boolean forced) throws ResourceUnavailableException, CloudException;
```

Comment:
@jburwell the `forced` flag has always existed, the `stopVirtualMachine` API accepts this and the query layer too has a method with this signature (https://github.com/apache/cloudstack/pull/1635/files#diff-0a1cd9df984252594918eec5acfed08cR3850) but it does not honour or pass this flag to the layers below, so effectively stopVM API with/without the `forced` flag has the same effect.